src/NServiceBus.AzureStoragePersistence/Timeout/Config/AzureStorageTimeoutPersistence.cs

```diff
@@ -27,14 +28,31 @@ internal AzureStorageTimeoutPersistence()
         protected override void Setup(FeatureConfigurationContext context)
         {
             //TODO: get rid of these statics
-            ServiceContext.CreateSchema = context.Settings.Get<bool>("AzureTimeoutStorage.CreateSchema");
-            ServiceContext.TimeoutDataTableName = context.Settings.Get<string>("AzureTimeoutStorage.TimeoutDataTableName");
-            ServiceContext.TimeoutManagerDataTableName = context.Settings.Get<string>("AzureTimeoutStorage.TimeoutManagerDataTableName");
+            var createIfNotExist = context.Settings.Get<bool>("AzureTimeoutStorage.CreateSchema");
+            var timeoutDataTableName = context.Settings.Get<string>("AzureTimeoutStorage.TimeoutDataTableName");
+            var timeoutManagerDataTableName = context.Settings.Get<string>("AzureTimeoutStorage.TimeoutManagerDataTableName");
+            var connectionString = context.Settings.Get<string>("AzureTimeoutStorage.ConnectionString");
+            var catchUpInterval = context.Settings.Get<int>("AzureTimeoutStorage.CatchUpInterval");
+            var partitionKeyScope = context.Settings.Get<string>("AzureTimeoutStorage.PartitionKeyScope");
+            var endpointName = context.Settings.EndpointName();
+            var hostDisplayName = context.Settings.GetOrDefault<string>("NServiceBus.HostInformation.DisplayName");
+            var timeoutStateBlobName = context.Settings.GetOrDefault<string>("AzureTimeoutStorage.TimeoutStateBlobName");
 
-            context.Container.ConfigureComponent<TimeoutPersister>(DependencyLifecycle.InstancePerCall)
-                .ConfigureProperty(tp => tp.ConnectionString, context.Settings.Get<string>("AzureTimeoutStorage.ConnectionString"))
-                .ConfigureProperty(tp => tp.CatchUpInterval, context.Settings.Get<int>("AzureTimeoutStorage.CatchUpInterval"))
-                .ConfigureProperty(tp => tp.PartitionKeyScope, context.Settings.Get<string>("AzureTimeoutStorage.PartitionKeyScope"));
+            var account = CloudStorageAccount.Parse(connectionString);
+
+            var timeoutTable = account.CreateCloudTableClient().GetTableReference(timeoutDataTableName);
+            if (createIfNotExist) timeoutTable.CreateIfNotExists();
+
+            var timeoutManagerTable = account.CreateCloudTableClient().GetTableReference(timeoutManagerDataTableName);
```

Comment:
@SimonCropp This is just an object construction w/o going to the physical account, so not expensive at all. Saying that, worth verification just in case anything has changed (/cc @dbelcham)